client/lib/domains/registration/availability-messages.js

```diff
@@ -258,4 +258,149 @@ function getAvailabilityNotice( domain, error, site ) {
 	};
 }
 
-export { getAvailabilityNotice };
+function getAvailabilityNoticeForTransfers(
+	domain,
+	status,
+	site,
+	rootDomain,
+	mappableStatus,
+	goToMapDomainStep
+) {
+	let message,
+		severity = 'error';
+
+	const isSubDomain = rootDomain && rootDomain !== domain;
+
+	switch ( status ) {
+		case domainAvailability.MAPPABLE:
+		case domainAvailability.TLD_NOT_SUPPORTED:
+			if ( isSubDomain ) {
+				message = translate(
+					'Transferring a subdomain is not possible. You can try to {{button}}map it{{/button}} instead ' +
+						'or transfer {{strong}}the root domain{{/strong}}.',
+					{
+						args: { rootDomain },
+						components: {
+							strong: <strong />,
+							button: <button onClick={ goToMapDomainStep } />,
+						},
+					}
+				);
+			} else {
+				const tld = getTld( domain );
+				message = translate(
+					"We don't support transfers for domains ending with {{strong}}.%(tld)s{{/strong}}, " +
+						'but you can {{button}}map it{{/button}} instead.',
+					{
+						args: { tld },
+						components: {
+							strong: <strong />,
+							button: <button onClick={ goToMapDomainStep } />,
+						},
+					}
+				);
+				severity = 'info';
+			}
+			break;
+		case domainAvailability.UNKNOWN:
+			if ( domainAvailability.MAPPABLE === mappableStatus ) {
+				message = translate(
+					"{{strong}}%(domain)s{{/strong}} can't be transferred. " +
+						'You can {{button}}manually connect it{{/button}} if you still want to use it for your site.',
+					{
+						args: { domain },
+						components: {
+							strong: <strong />,
+							button: <button onClick={ goToMapDomainStep } />,
+						},
+					}
+				);
+				severity = 'info';
+				// The break is here intentionally so we can fallback to the default
+				break;
+			}
+		default:
+			( { message, severity } = getAvailabilityNotice( domain, status, site ) );
+	}
+
+	return {
+		message,
+		severity,
+	};
+}
+
+function getOfferAvailabilityForSearchResults(
+	lastDomainSearched,
+	domain,
+	rootDomain,
+	isNextDomainFree,
+	handleAddMapping,
+	domainsWithPlansOnly,
+	isSiteOnPaidPlan,
+	domainMapCost,
+	tldNotSupported
+) {
+	let offer, domainUnavailableMessage;
+	const components = { button: <button onClick={ handleAddMapping } />, small: <small /> };
```

Comment:
I'm guessing you did it this way because of some a11y linting, but this introduces a style that is not consistent with the other notices and seems off. The font size is slightly different and the cursor does not change to a pointer:
![2018-02-15 05_01_09](https://user-images.githubusercontent.com/3392497/36239861-4581626e-120d-11e8-8e8c-d2048b3f6ae0.gif)